### PR TITLE
Use new models for all sensors

### PR DESCRIPTION
**Pull request by jschlyter**
## Release Notes

- **New Features**
	- Introduced methods to update car information, battery, and odometer data for improved accuracy.
	- Added a new field for software version timestamp in car information data.

- **Improvements**
	- Simplified sensor data retrieval logic for better performance and clarity.
	- Enhanced error handling in API calls for more robust operation.

- **Bug Fixes**
	- Adjusted expected data types in tests to align with recent changes in the data model.

### CODE REVIEW

File: custom_components/polestar_api/polestar.py [hunk 1]
```diff
@@ -49,20 +49,130 @@ def get_device_info(self) -> DeviceInfo:
...
+                )
+            else:
+                estimate_full_charge_range = None
+
+            estimated_fully_charged_time = (
+                datetime.now().replace(second=0, microsecond=0)
+                + timedelta(minutes=round(data.estimated_charging_time_to_full_minutes))
+                if data.estimated_charging_time_to_full_minutes
+                else "Not charging"
+            )
```
[the remark this answers is not in the document]
Comment (jschlyter) [reply custom_components/polestar_api/polestar.py:109]: Will change to return string as datetime is misparse by statistics

File: custom_components/polestar_api/polestar.py [hunk 2]
```diff
@@ -49,20 +49,134 @@ def get_device_info(self) -> DeviceInfo:
...
+                    "registration_number": data.registration_no,
+                    "registration_date": data.registration_date,
+                    "factory_complete_date": data.factory_complete_date,
+                    "model_name": data.model_name,
+                    "software_version": data.software_version,
+                    "software_version_release": data.software_version_timestamp,
+                    "battery_capacity": battery_capacity,
+                    "torque": torque,
+                }
+            )
```
[the remark this answers is not in the document]
Comment (jschlyter) [reply custom_components/polestar_api/polestar.py:85]: Should be handled by caller.

**Comment by jschlyter** [comment]
Perhaps @BJReplay can take a look? Once this is moved, my plan is to move the pypolestar library to pypi.

**Comment by BJReplay** [comment]
> Perhaps @BJReplay can take a look?

Will do. Currently in the "My delivery is delayed again loop" 🤣 hoping that it is not an infinite loop as the delivery estimate has gone from fixed dates in November to early December to August, and I am wondering exactly which year! August next, or will they build a time machine and deliver the car last August?

**Comment by BJReplay** [comment]
> Perhaps @BJReplay can take a look?

Again, just running this in a Dev Container, merged into latest (1.10.0) and also #254 against an undelivered vehicle.

As per comments on #254, ruff made some complaints about the touched modules, but none in this case on the touched lines of code.  I can't see any changes on most of the sensors as I have an undelivered car, but yep, torque is definitely in Nm now :)

Nice to see build date moved to an attribute as planned.  The state property of the sensor is a lot cleaner as a result of this change, as is the dictionary of sensor definitions, and they'll be easier to maintain.

**Pull request merged**